osf_tests/test_institution.py

```diff
@@ -103,3 +106,65 @@ def test_deactivated_institution_in_all_institutions(self):
         institution.deactivated = timezone.now()
         institution.save()
         assert institution in Institution.objects.get_all_institutions()
+
+    def test_deactivate_institution(self):
+        institution = InstitutionFactory()
+        with mock.patch.object(
+                institution,
+                '_send_deactivation_email',
+                return_value=None
+        ) as mock__send_deactivation_email:
+            institution.deactivate()
+            assert institution.deactivated is not None
+            assert mock__send_deactivation_email.called
+
+    def test_reactivate_institution(self):
+        institution = InstitutionFactory()
+        institution.deactivated = timezone.now()
+        institution.save()
+        institution.reactivate()
+        assert institution.deactivated is None
+
+    @mock.patch('website.mails.send_mail', return_value=None, side_effect=mails.send_mail)
```

Comment:
Using `side_effect` to ensure that the template are rendered during testing. This catches malformed templates or ones with bad / unsupported encoding.

Reply:
nit: consider also mocking `website.mails.settings.USE_EMAIL` to False, just in case local settings are weird.